crates/ra_hir/src/expr.rs

```diff
@@ -633,13 +648,52 @@ impl ExprCollector {
                 let exprs = e.exprs().map(|expr| self.collect_expr(expr)).collect();
                 self.alloc_expr(Expr::Tuple { exprs }, syntax_ptr)
             }
+            ast::ExprKind::Literal(e) => {
+                let child = if let Some(child) = e.literal_expr() {
+                    child
+                } else {
+                    return self.alloc_expr(Expr::Missing, syntax_ptr);
+                };
+
+                let lit = match child.flavor() {
+                    LiteralFlavor::IntNumber { suffix } => {
+                        let known_name = suffix
+                            .map(|s| Name::new(s))
+                            .and_then(|name| UncertainIntTy::from_name(&name));
+
+                        if let Some(kn) = known_name {
+                            Literal::Int(Default::default(), kn)
+                        } else {
+                            Literal::Int(Default::default(), UncertainIntTy::Unknown)
+                        }
```

Comment:
this could just be `Literal::Int(Default::default(), known_name.unwrap_or(UncertainIntTy::Unknown))`, right?

Reply:
Yes!